React/CoreModules/RCTDeviceInfo.mm

```diff
@@ -202,16 +213,20 @@ - (void)interfaceFrameDidChange
 - (void)_interfaceFrameDidChange
 {
   NSDictionary *nextInterfaceDimensions = RCTExportedDimensions(_moduleRegistry, _bridge);
-
-  if (!([nextInterfaceDimensions isEqual:_currentInterfaceDimensions])) {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-    [[_moduleRegistry moduleForName:"EventDispatcher"] sendDeviceEventWithName:@"didUpdateDimensions"
-                                                                          body:nextInterfaceDimensions];
-#pragma clang diagnostic pop
+  UIApplicationState appState = [RCTSharedApplication() applicationState];
+    
+  BOOL isActive = appState == UIApplicationStateActive;
+  // update and publish the even only when the app is in active state
+  if (!([nextInterfaceDimensions isEqual:_currentInterfaceDimensions]) && isActive) {
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        [[_moduleRegistry moduleForName:"EventDispatcher"] sendDeviceEventWithName:@"didUpdateDimensions"
+                                                                              body:nextInterfaceDimensions];
+          // We only want to track the current _currentInterfaceOrientation and _isFullscreen only 
+          // when it happens and only when it is published, thus it has to happen inside the if condition.
```

Comment:
```suggestion
          // We only want to track the current _currentInterfaceOrientation and _isFullscreen only 
          // when it happens and only when it is published.
```

instead of it happens, we should tell what exactly is happening? This is not clear to me.